src/env.rs

```diff
@@ -0,0 +1,57 @@
+use std::env;
+use {Error, Result};
+
+/// Clear the environment of all name-value pairs.
+///
+/// On platforms where libc provides `clearenv()`, it will be used. libc's
+/// `clearenv()` is documented to return an error code but not set errno; if the
+/// return value indicates a failure, this function will return
+/// `Error::UnsupportedOperation`.
+///
+/// On platforms where libc does not provide `clearenv()`, a fallback
+/// implementation will be used that iterates over all environment variables and
+/// removes them one-by-one.
+///
+/// # Safety
+///
+/// This function is not threadsafe and can cause undefined behavior in
+/// combination with `std::env` or other program components that access the
+/// environment. See, for example, the discussion on `std::env::remove_var`; this
+/// function is a case of an "inherently unsafe non-threadsafe API" dealing with
+/// the environment.
+///
+///  The caller must ensure no other threads access the process environment while
+///  this function executes and that no raw pointers to an element of libc's
+///  `environ` is currently held. The latter is not an issue if the only other
+///  environment access in the program is via `std::env`, but the requirement on
+///  thread safety must still be upheld.
+pub unsafe fn clearenv() -> Result<()> {
+    let ret;
```

Comment:
As a matter of style, I suggest:
```rust
let ret = cfg_if! {
    if something {
        libc::clearenv()
    } else {
        0
    }
};
```

Reply:
I tried that first, but it looks like that's not allowed with `cfg_if!`. It generates a lot of compilation errors; the most relevant looks like:

```
error: macro expansion ignores token `$crate` and any following
  --> <::cfg_if::cfg_if macros>:23:37
   |
23 |       { @ __identity $ ($ tokens) * } $ crate :: cfg_if !
   |                                       ^^^^^^^
   |
  ::: src/env.rs:28:15
   |
28 |       let ret = cfg_if! {
   |  _______________-
29 | |         if #[cfg(any(target_os = "fuchsia",
30 | |                      target_os = "wasi",
31 | |                      target_env = "wasi",
...  |
43 | |         }
44 | |     };
   | |     -
   | |     |
   | |_____caused by the macro expansion here
   |       help: you might be missing a semicolon here: `;`
   |
   = note: the usage of `cfg_if!` is likely invalid in expression context
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```